### PR TITLE
Add static qualifiers to json creators for onCompletion enum serialization.

### PR DESCRIPTION
**Pull request by egieseke**
Add static qualifiers to json creators for onCompletion enum serialization.

### CODE REVIEW

File: src/main/java/com/algorand/algosdk/transaction/Transaction.java [hunk 1]
```diff
@@ -1185,9 +1185,13 @@ public OnCompletion String(int value) {
             return null;
         }
 
+        public int getSerializedValue() {
```
Comment (winder) [inline src/main/java/com/algorand/algosdk/transaction/Transaction.java:1188]: Why have a separate function here?

**Comment by winder** [comment]
@egieseke thanks! Could you add the unit test from #220 to `TestTransaction.java` as well?

File: src/main/java/com/algorand/algosdk/transaction/Transaction.java (hunk 1 above)
Comment (egieseke) [reply src/main/java/com/algorand/algosdk/transaction/Transaction.java]: Removed separate function.

Comment (egieseke) [reply src/main/java/com/algorand/algosdk/transaction/Transaction.java]: > @egieseke thanks! Could you add the unit test from #220 to `TestTransaction.java` as well?

Added unit test for Transaction and AppTransaction to TestTransaction.java.

Comment (egieseke) [reply src/main/java/com/algorand/algosdk/transaction/Transaction.java]: Removed this function and retested after removing JSONCreator annotation from string deserializer.

**Review by winder — APPROVED** [review]

**Pull request merged**